projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/listeners/Function.java

```diff
@@ -982,6 +982,20 @@ private static int getAlign(long argument) {
         return (int) argument & (Long.SIZE - 1);
     }
 
+    private static Type getExtractedElementType(Type type) {
+        Type valType = type;
+        if (valType instanceof PointerType) {
```

Comment:
Thank you for pointing that out. I was missing that I am getting completely wrong types from `types.get(i++);` in the case of a forward reference. `types.get(args[i++]);` retrieves the correct type for me which is always a vector type as specified in the docs. I will change my fix accordingly. However, I am wondering if there exist cases where `types.get(i++);` would retrieve the correct type rather than `types.get(args[i++]);`.